### ISSUE TITLE
Can't add funds if an event is archived

### ISSUE THREAD

**Issue by poohlaga** [opencollective/opencollective#6619]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - Browser: [e.g. chrome, safari]
 - Version: [e.g. 22]

**Smartphone (please complete the following information):**
 - Browser: [e.g. stock browser, safari]
 - Version: [e.g. 22]

**Additional context**

<details>
  <summary>Triage Template (core team only)</summary>

  This template is for members of the team to triage for prioritisation. For more guidance see https://www.loom.com/share/369ab467fbc64dec848085d38ff57ca0:

  P1 high frequency, high impact
  P2 low frequency, high impact
  P3 high frequency, low impact
  P4 low frequency, low impact

  Examples of high impact - a problem affects users during an essential process (expenses and payments > onboarding and registration > contributing) with no workaround.

  High frequency - >10% of users affected (measured as a proportion of total potential users for this case).
</details>

**Comment by Betree** [comment]
Found the bug; it's actually affecting all events & projects - not only archived. Pushing a fix.

**Comment by Betree** [comment]
Project's URL: https://opencollective.com/ifcjs/events/ifcjs-crash-course

I don't consider this a bug; it's the expected behavior. Archiving an account means you're freezing all its activities - especially the financial ones. The error message could be improved, though.

I suggest either adding the funds to the parent collective or asking the admins to un-archive the project.

**Comment by BenJam** [comment]
> I don't consider this a bug

I agree, in this instance i would propose adding the funds to the parent Collective (where they would be allocated when the event was archived anyways). 

That said, @Betree I can see why this behaviour would have been requested: because the contributor would, understandably, expect to be credited as a contributor to the event if, say, they were sponsoring. 

In which case, perhaps the best behaviour here is to allow the contributor recognition but allocate the contribution to the parent Collective, which might require a confirmation dialogue:

> This <event/project> is Archived, the Contributor will be recognised, but the contribution will be allocated to <parent collective>

**Comment by poohlaga** [comment]
It also makes me think this is just a work around for an end date on events to stop registrations. @BenJam not sure if that is a feature request for events that has been noted?

**Comment by Betree** [comment]
> It also makes me think this is just a work around for an end date on events to stop registrations. @BenJam not sure if that is a feature request for events that has been noted?

This is how it works today: apart from admins, people cannot buy tickets after an event's end date.

Note that we got the feedback that some want to keep registrations open after the end date, but I disagree with that as I can't think of one event-managing platform allowing it: https://github.com/opencollective/opencollective/issues/6367
